pkg/doc/verifiable/credential_ldp_test.go

```diff
@@ -208,6 +211,191 @@ func TestNewCredentialFromLinkedDataProof_JSONLD_Validation(t *testing.T) {
+    "type": "CredentialStatusList2017"
+  }
+}`
+
+	pubKey, privKey, err := ed25519.GenerateKey(rand.Reader)
+	r.NoError(err)
+
+	sigSuite := ed25519signature2018.New(
+		suite.WithSigner(getEd25519TestSigner(privKey)),
+		suite.WithVerifier(ed25519signature2018.NewPublicKeyVerifier()))
+
+	ldpContext := &LinkedDataProofContext{
+		SignatureType:           "Ed25519Signature2018",
+		SignatureRepresentation: SignatureProofValue,
+		Suite:                   sigSuite,
+		VerificationMethod:      "did:example:123456#key1",
+	}
+
+	vc, _, err := NewCredential([]byte(vcJSON))
+	r.NoError(err)
+
+	err = vc.AddLinkedDataProof(ldpContext)
+	r.NoError(err)
+
+	vcBytes, err := json.Marshal(vc)
+	r.NoError(err)
+
+	vcWithLdp, _, err := NewCredential(vcBytes,
+		WithEmbeddedSignatureSuites(sigSuite),
+		WithPublicKeyFetcher(SingleKey(pubKey, kms.ED25519)),
+		WithStrictValidation())
+	r.NoError(err)
+	r.Equal(vc, vcWithLdp)
+	r.NotNil(vcWithLdp)
+
+	// Drop https://trustbloc.github.io/context/vc/examples-v1.jsonld context where
+	// SupportingActivity and CredentialStatusList2017 are defined.
+	vcMap, err := toMap(vcBytes)
+	r.NoError(err)
+
+	vcMap["@context"] = baseContext
+	vcBytes, err = json.Marshal(vcMap)
+	r.NoError(err)
+
+	vcWithLdp, _, err = NewCredential(vcBytes,
+		WithEmbeddedSignatureSuites(sigSuite),
+		WithPublicKeyFetcher(SingleKey(pubKey, kms.ED25519)),
+		WithStrictValidation())
+	r.Error(err)
+	r.EqualError(err, "decode new credential: check embedded proof: check linked data proof: ed25519: invalid signature")
+	r.Nil(vcWithLdp)
+
+	// Use extra context.
+	vcWithLdp, _, err = NewCredential(vcBytes,
+		WithEmbeddedSignatureSuites(sigSuite),
+		WithPublicKeyFetcher(SingleKey(pubKey, kms.ED25519)),
+		WithExtraJSONLDContexts("https://trustbloc.github.io/context/vc/examples-v1.jsonld"),
+		WithStrictValidation())
+	r.NoError(err)
+	r.NotNil(vcWithLdp)
+
+	// Use extra context.
+	vcWithLdp, _, err = NewCredential(vcBytes,
+		WithEmbeddedSignatureSuites(sigSuite),
+		WithPublicKeyFetcher(SingleKey(pubKey, kms.ED25519)),
+		WithExtraJSONLDContexts("https://trustbloc.github.io/context/vc/examples-v1.jsonld"),
+		WithStrictValidation())
+	r.NoError(err)
+	r.NotNil(vcWithLdp)
+
+	// Use extra in-memory context.
+	reader, err := ld.DocumentFromReader(strings.NewReader(`
+{
+    "@context": {
+      "@version": 1.1,
+  
+      "id": "@id",
+      "type": "@type",
+      
+      "ex": "https://example.org/examples#",
+
+      "CredentialStatusList2017": "ex:CredentialStatusList2017",
+      "DocumentVerification": "ex:DocumentVerification",
+      "SupportingActivity": "ex:SupportingActivity"
+    }
+}
+`))
+	r.NoError(err)
+
+	loader := CachingJSONLDLoader()
+	loader.AddDocument("http://localhost:8652/dummy.jsonld", reader)
+
+	vcWithLdp, _, err = NewCredential(vcBytes,
+		WithEmbeddedSignatureSuites(sigSuite),
+		WithPublicKeyFetcher(SingleKey(pubKey, kms.ED25519)),
+		WithExtraJSONLDContexts("http://localhost:8652/dummy.jsonld"),
```

Comment:
I think we can add another option for in-memory context, e.g. `WithExtraInMemoryJSONLDContext()`, where we will pass just in-memory context (and won't pass this dummy context URL like `http://localhost:8652/dummy.jsonld`). Under the hood, it will generate this URL (with e.g. UUID identifier).

Reply:
right - this is similar to the embedded/inline context https://www.w3.org/TR/json-ld11/#example-7-in-line-context-definition

Reply:
Filed https://github.com/hyperledger/aries-framework-go/issues/1758